warehouse/static/js/warehouse/controllers/statuspage_controller.js

```diff
@@ -0,0 +1,258 @@
+          "created_at": "2014-05-14T14:34:20.036-06:00",
+          "display_at": "2014-05-14T14:34:20.036-06:00",
+          "id": "drs62w8df6fs",
+          "incident_id": "k7mf5z1gz05c",
+          "status": "in_progress",
+          "updated_at": "2014-05-14T14:34:20.036-06:00"
+        },
+        {
+          "body": "We will be performing rolling upgrades to our web tier with a new kernel version so that Heartbleed will stop making us lose sleep at night. Increased load and latency is expected, but the app should still function appropriately. We will provide updates every 30 minutes with progress of the reboots.",
+          "created_at": "2014-05-14T14:27:18.845-06:00",
+          "display_at": "2014-05-14T14:27:18.845-06:00",
+          "id": "z40y7398jqxc",
+          "incident_id": "k7mf5z1gz05c",
+          "status": "scheduled",
+          "updated_at": "2014-05-14T14:27:18.845-06:00"
+        }
+      ],
+      "monitoring_at": null,
+      "name": "Web Tier Recycle",
+      "page_id": "2p66nmmycsj3",
+      "resolved_at": null,
+      "scheduled_for": "2014-05-14T14:30:00.000-06:00",
+      "scheduled_until": "2014-05-14T16:30:00.000-06:00",
+      "shortlink": "http://stspg.dev:5000/Q0G",
+      "status": "in_progress",
+      "updated_at": "2014-05-14T14:35:12.258-06:00"
+    }
+  ]
+};
+
+const mockFetch = function(endpoint){
+  if (endpoint.indexOf('status.json') !== -1) {
+    return new Promise((resolve, reject) => resolve(mockStatusResponse));
+  } else if (endpoint.indexOf('summary.json') !== -1) {
+    return new Promise((resolve, reject) => resolve(mockSummaryResponse));
+  }
+};
+
+export default class extends Controller {
+  static targets = ["title", "components", "incidents"];
+  incidentLimit = 3;
+  statusPageDomain = null;
+  // Enable mock data
+  mock = true;
+
+  jsUcfirst(string) {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+  }
+
+  _formatUtcDateTime(time) {
+    const date = new Date(time);
+    return `${date.getUTCFullYear()}-${date.getUTCMonth()}-${date.getUTCDate()} ${date.getUTCHours()}:${date.getUTCMinutes()}:${date.getUTCSeconds()} UTC`;
+  }
+
+  connect() {
+    if (typeof window.fetch !== "undefined") {
+      this.statusPageDomain = this.element.getAttribute("data-statuspage-domain");
+      if (this.statusPageDomain === null) {
+        return;
+      }
+
+      if (this.mock) {
+        fetch = mockFetch;
+      }
+
+      fetch(`${this.statusPageDomain}/api/v2/status.json`).then((response) => {
+        if (this.mock) {
+          return response;
+        }
+        return response.json();
+      }).then((json) => {
+        const description = json.status.description;
+        // If we get something other than "none", this is not normal.
+        // if (json.status.indicator !== "none") {
+          return json.status.indicator;
+        // }
+      }).then((indicator) => {
+        fetch(`${this.statusPageDomain}/api/v2/summary.json`)
```

Comment:
I'm unfamiliar with the status page API, do we need to make two requests?

As it stands we're making a request to `status` and then another one to `summary`, but seems they're independent since it's being done regardless of the `indicator` or `description`. Also in your mocked responses the `status` response is fully included in the `summary` response.

Could you clarify please?